src/__tests__/userEvent/userEvent.press.test.tsx

```diff
@@ -0,0 +1,188 @@
+        pointerEvents="box-only"
+        testID="pressable"
+      />
+    );
+
+    userEvent.press(screen.getByTestId('pressable'));
+
+    expect(mockOnPress).toHaveBeenCalled();
+  });
+
+  test('crawls up in the tree to find an element that responds to touch events', () => {
+    const mockOnPress = jest.fn();
+
+    render(
+      <Pressable onPress={mockOnPress}>
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'));
+
+    expect(mockOnPress).toHaveBeenCalled();
+  });
+
+  test('calls onPressIn', () => {
+    const mockOnPressIn = jest.fn();
+
+    render(
+      <Pressable onPressIn={mockOnPressIn}>
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'));
+
+    expect(mockOnPressIn).toHaveBeenCalled();
+  });
+
+  test('calls onPressOut', () => {
+    const mockOnPressOut = jest.fn();
+
+    render(
+      <Pressable onPressOut={mockOnPressOut}>
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'));
+
+    expect(mockOnPressOut).toHaveBeenCalled();
+  });
+
+  test('does not call onLongPress for pressDuration of 0', () => {
+    const mockOnLongPress = jest.fn();
+
+    render(
+      <Pressable onLongPress={mockOnLongPress}>
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'));
+
+    expect(mockOnLongPress).not.toHaveBeenCalled();
+  });
+
+  test('calls onLongPress for a duration greater than default onLongPressDelay', () => {
+    const mockOnLongPress = jest.fn();
+
+    render(
+      <Pressable onLongPress={mockOnLongPress}>
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'), { pressDuration: 500 });
+
+    expect(mockOnLongPress).toHaveBeenCalled();
+  });
+
+  test('calls onLongPress when duration is greater than specified onLongPressDelay', () => {
+    const mockOnLongPress = jest.fn();
+    const mockOnPress = jest.fn();
+
+    render(
+      <Pressable
+        delayLongPress={1000}
+        onLongPress={mockOnLongPress}
+        onPress={mockOnPress}
+      >
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'), { pressDuration: 500 });
+
+    expect(mockOnLongPress).not.toHaveBeenCalled();
+    expect(mockOnPress).toHaveBeenCalledTimes(1);
+
+    userEvent.press(screen.getByText('press me'), { pressDuration: 1000 });
+
+    expect(mockOnLongPress).toHaveBeenCalled();
+    expect(mockOnPress).toHaveBeenCalledTimes(1);
+  });
+
+  test('does not calls onPress when onLongPress is called', () => {
+    const mockOnLongPress = jest.fn();
+    const mockOnPress = jest.fn();
+
+    render(
+      <Pressable onLongPress={mockOnLongPress} onPress={mockOnPress}>
+        <Text>press me</Text>
+      </Pressable>
+    );
+    userEvent.press(screen.getByText('press me'), { pressDuration: 500 });
```

Comment:
I'm wondering whether having separate `userEvent.longPress` would be more ergonomic API than passing `pressDuration` above certain threshold, which might not be obvious.

As a quite inexact metaphor, the original User Event library has [both `click` and `dblClick`](https://testing-library.com/docs/user-event/convenience#clicks) (and `tripleClick` as well ;-)).

```suggestion
    userEvent.longPress(screen.getByText('press me'));
```

Reply:
Yes I was thinking initially of making a press and longPress API but the thing is long press only exists on composite side of components. To implement it, we'd need to look at the value of the delayLongPress prop on the composite component and I wanted to only use host components to keep the implementation simple and closer to what happens at host level

Reply:
Perhaps we could structure in the way where `press` is having a very brief duration (e.g. 0 ms, or 50 ms, etc), while `longPress` would accept `duration` option with the default value of 500 ms (the default for `Pressable`).

That would result in quite intuitive tests:

```
```

Reply:
Yes that could work. The only thing that bothers me is that when using the API, longPress may be more interpreted as "it calls the onLongPressProp" rather than user is pressing for a long time. And if the delayLongPress is greater than 500ms then it won't be called which may be perceived as a bug. The major advantage of this is that most users may not know what the default delay is for react to consider it to be a long press. In the end it's a question of how much we want to abstract this. I'm wondering if it wouldn't be better to start with only the press API and then based on user feedback see if there is a need for a longPress API

Reply:
I think `longPress` is a much rare case than regular `press`, so I would optimise the API for regular `press`. Then for long press most of the users would probably stick to the default delay, so in the next order I would optimise the API for them. Only final, there will be people who use long press with custom duration, so IMO they are the edge case of an edge case. 

Therefore, making their API a bit more complex yet still possible seems like a good tradeoff, as most of the users will have simple and intuitive API.

BTW afaik we are neither calling `onPress` nor `onLongPress` directly. If users want direct `onLongPress` then they can use `fireEvent`